torch/_refs/__init__.py

```diff
@@ -781,11 +889,82 @@ def cat(tensors: TensorSequenceType, dim: int = 0) -> TensorLikeType:
     return prims.concatenate(tensors, _dim)
 
 
+def chunk(a: TensorLikeType, chunks: int, dim: int = 0) -> Tuple[TensorLikeType, ...]:
+    if chunks <= 0:
+        msg = "Expected at least one chunk, but got {0}!".format(chunks)
+        raise ValueError(msg)
+
+    dim = utils.canonicalize_dim(a.ndim, dim)
+    length = a.shape[dim]
+    chunk_size = math.ceil(length / chunks)
+    full_chunks = math.floor(length / chunk_size)
+    tail_chunk_size = length % chunk_size
+
+    result = []
+    for i in range(full_chunks):
+        result.append(narrow(a, dim, i * chunk_size, chunk_size))
+
+    if tail_chunk_size != 0:
+        result.append(narrow(a, dim, full_chunks * chunk_size, tail_chunk_size))
+
+    return tuple(result)
+
+
+# Note: flatten, unlike prim.collapse and prim.collapse_view has an inclusive end_dim
+def flatten(a: TensorLikeType, start_dim: int = 0, end_dim: int = -1) -> TensorLikeType:
+    start_dim = utils.canonicalize_dim(a.ndim, start_dim)
+    end_dim = utils.canonicalize_dim(a.ndim, end_dim) + 1
+
+    # Tries to take a view
+    # TODO: we could look at directing collapse_view to skip its meta function here
+    new_shape, new_strides = prims._collapse_view_helper(a, start_dim, end_dim)
+    if new_shape is not None:
+        return prims.collapse_view(a, start_dim, end_dim)
+
+    # Makes a copy if it can't make a view
+    result = prims.collapse(a, start_dim, end_dim)
+    return result
+
+
+def flip(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
+    dims = utils.canonicalize_dims(a.ndim, dims)  # type: ignore[assignment]
+    return prims.rev(a, dims)
+
+
+def narrow(a: TensorLikeType, dim: int, start: int, length: int) -> TensorLikeType:
+    dim = utils.canonicalize_dim(a.ndim, dim)
+    return prims.slice_in_dim(a, start, start + length, axis=dim)
+
+
 def permute(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
     _permutation = utils.canonicalize_dims(a.ndim, dims)
     return prims.transpose(a, _permutation)
 
 
+@out_wrapper
+def stack(tensors: TensorSequenceType, dim: int = 0) -> TensorLikeType:
+    tensors = tuple(unsqueeze(a, dim) for a in tensors)
```

Comment:
this is a very inefficient decomposition that's hard to make efficient again in the backend, a better one is catting the tensor and viewing the result. 
Also, where are the size checks for inputs?

Reply:
The size check is handled by cat but we can add one here, too

Reply:
Added a note for performance

Reply:
No, size check in the cat makes sure tensors are ok for `cat`, but stack is much more restrictive - it requires all inputs of the same size (including stacked dimension).

Reply:
per offline discussion: @ngimel is right but we actually are OK without another check here because this does not implement the fast path for stack where it's catted and then viewed (there is a comment to add this)